### PR TITLE
Rework ResizableTable column rendering

### LINKED ISSUE #4001: Column resizing is laggy / unresponsive
## Bug description

The column resizing (which is a pretty useful feature) is quite laggy (see recording below), the experience isn't great.

## Expected behavior

I would expect columns to resize very smoothly. In addition it'd be great if we could add instrumentation for when a user uses this feature so we can later improve it.

## How to reproduce

See recording.

## Environment

- [X] PostHog Cloud. Latest.
- [ ] self-hosted PostHog, version/commit: _please provide_

### PR DESCRIPTION
**Pull request by samwinslow**
## Changes

Fixes #4001, although needs further testing.

`ResizableTable` now renders the antd table separate from the header, which is a custom component called `VirtualTableHeader`. The previous approach used resizable components within the actual table, which forced the whole table to re-render on resize.

The header now controls the size of the columns via imperative manipulation of `style.width` on the `<colgroup>` rendered by antd. This avoids re-rendering the antd table because its props do not change.

### What still needs to be done:

Resize the whole table when its wrapper resizes.

Currently, the column sizes are set on first render, and will not grow if the user expands the window. We should probably keep track of which columns were user-resized and which were auto-sized and recalculate sizes for the auto-sized columns.

https://user-images.githubusercontent.com/4645779/115152025-73222980-a03d-11eb-871d-9c0401052fd0.mp4

This also means the table won't grow if it is mounted too soon after other animations take place on the page, for example, the sidebar-hiding animation:

https://user-images.githubusercontent.com/4645779/115152017-6d2c4880-a03d-11eb-9a15-14d6eaded126.mp4

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

**Comment by samwinslow** [comment]
@paolodamico can we sync up soon to see if this addresses the performance issues you found? I would also love your input on making the whole component expand on window resize.

**Comment by paolodamico** [comment]
Hey @samwinslow! Did an initial pass and here are my comments,
1. Functionality-wise this looks quite promising. Performance issues seem gone and resizing seems to be quite responsive now.
1. Styling-wise the headers look a bit off, they have a lot of extra height, seems like we need to add a background, and the column drag/separator seems to dark or thick. I would recommend emulating Ant's design as much as possible. Happy to help here too if you want.
1. In terms of resizing based on external factors, I would intuitively try to resize proportionally when changing the window size instead of keeping fixed lengths of columns, but this might be one of those things where we need to actually use it to make sure it's as intuitive as possible.
1. I would particularly try to consider the case of when the navigation bar is hidden, as this is a base functionality (it could be as simple as triggering a window resize event [based on point above], so the table width gets recomputed. As an alternative approach, we could consider opening the navigation bar on top of the content, without causing a resize / relocation of the main content.

**Comment by samwinslow** [comment]
@paolodamico – Addressed most of this, with the exception of styling the height; `react-resizable` components have fixed height. There's a chance I can strip out `react-resizable` and recreate its functionality with browser-native resizable divs and `ResizeObserver` (now that I know about that API) — but in effort to keep this PR down to a manageable size maybe that can be handled later.

**Review by EDsCODE — COMMENTED** [review]
The resizing is much smoother! Did a quick QA and noticed a few things

1. Sessions table isn't resizable. It seems like it's stuck in a loop updating because the draggable bars are shaking (they look like they're wobbling between positions)
2. I think this was the UX before, but would it be possible to make the table so that it always fills up the width even as you resize. Right now you could pull all the columns to one side and make the table awkwardly small

https://user-images.githubusercontent.com/13127476/115731901-81ca5280-a355-11eb-8770-7ee1f11ef34d.mov

**Comment by samwinslow** [comment]
Going to look into using [react-data-grid](https://github.com/adazzle/react-data-grid). I think hacking Antd components to this extent is becoming a lost cause.

**Review by paolodamico — COMMENTED** [review]
Headers looking much better! Looks like @EDsCODE is now reviewing this, but lmk if you still need help with this.

File: frontend/src/lib/components/ResizableTable/VirtualTableHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    layoutEffect,
+    minColumnWidth,
+}: VirtualTableHeaderProps<RecordType>): JSX.Element {
+    const maxColumnWidth = minColumnWidth * 12
+    const height = 60
+    useLayoutEffect(() => (typeof layoutEffect === 'function' ? layoutEffect() : undefined))
+    return (
+        <div className="resizable-virtual-table-header">
+            <div className="left-spacer" style={{ width: ANTD_EXPAND_BUTTON_WIDTH }} />
+            {columns.map(({ /*ellipsis, key,*/ title, width }, index) => (
```
Comment (EDsCODE) [inline frontend/src/lib/components/ResizableTable/VirtualTableHeader.tsx:76]: Did you want to remove this comment?

**Review by EDsCODE — APPROVED** [review]
Tested again too. I did end up with the sessions problem briefly but wasn't able to reproduce it properly. I think it's good to merge this and we'll keep an eye on that. I couldn't figure out what might've been causing that stuttering

**Pull request merged**